__tests__/utils/shortString.test.ts

```diff
@@ -20,11 +20,11 @@ describe('shortString', () => {
   });
 
   test('should convert hex number string to string', () => {
-    expect(decodeShortString('0x68656c6c6f')).toMatchInlineSnapshot(`"hello"`);
+    expect(JSON.stringify(decodeShortString('0x68656c6c6f'))).toBe('"hello"');
   });
 
   test('should convert decimal number string to string', () => {
-    expect(decodeShortString('448378203247')).toMatch(`hello`);
+    expect(JSON.stringify(decodeShortString('448378203247'))).toBe('"hello"');
```

Comment:
Should we maybe add a new test case/cases with the example you mentioned (0X) that caused the bug?

Reply:
Not required as I left it as '0X' in decodeShortString and modified the test to detect hidden characters.
This test will not pass if there is an issue.

Reply:
On the other thought, I added it to have more transparent test.
ab4d341c3df6000ff4d0e14cd4d8788a8e07cb7f